### PR TITLE
Modify rule S3923: LaYC identical branches

### PR DESCRIPTION
**Pull request by frederic-tingaud-sonarsource**

### CODE REVIEW

File: rules/S3923/abap/rule.adoc [hunk 1]
```diff
@@ -1,8 +1,6 @@
 == Why is this an issue?
 
-Having all branches in a ``++CASE++`` or ``++IF++`` chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be a ``++CASE++``/``++IF++`` chain at all.
-
-=== Noncompliant code example
+Having all branches in a `CASE` or `IF` chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be a `CASE`/`IF` chain at all.
```
Comment (amelie-renard-sonarsource) [inline rules/S3923/abap/rule.adoc:3]:
```suggestion
Having all branches in a `CASE` or `IF` chain with the same implementation is odd. Either a copy-paste error was made, and something different should be executed, or there shouldn't be a `CASE`/`IF` chain.
```
To be less harsh?
(+Grammarly)
(and same comment for the other languages)

File: rules/S3923/abap/rule.adoc [hunk 2]
```diff
@@ -1,8 +1,6 @@
...
 
 [source,abap]
```
Comment (amelie-renard-sonarsource) [inline rules/S3923/abap/rule.adoc:5]: Could you find a sentence to introduce this code?
(and same comment for the other languages)

File: rules/S3923/plsql/rule.adoc [hunk 3]
```diff
@@ -1,8 +1,6 @@
 == Why is this an issue?
 
-Having all branches in a ``++CASE++`` or ``++IF++``/``++ELSIF++`` chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be a ``++CASE++``/``++IF++``/``++ELSIF++`` chain at all.
-
-=== Noncompliant code example
+Having all branches in a `CASE` or `IF`/`ELSIF` chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be a `CASE`/`IF`/`ELSIF` chain at all.
```
Comment (amelie-renard-sonarsource) [inline rules/S3923/plsql/rule.adoc:3]:
```suggestion
Having all branches in a `CASE` or `IF`/`ELSIF` chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be a `CASE` or `IF`/`ELSIF` chain at all.
```

File: rules/S3923/ruby/rule.adoc [hunk 4]
```diff
@@ -1,8 +1,6 @@
 == Why is this an issue?
 
-Having all branches in a ``++case++`` or ``++if++`` chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be a ``++case++``/``++if++`` chain at all.
-
-=== Noncompliant code example
+Having all branches in a `case` or `if` chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be a `case`/`if` chain at all.
```
Comment (amelie-renard-sonarsource) [inline rules/S3923/ruby/rule.adoc:3]:
```suggestion
Having all branches in a `case` or `if` chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be a `case` or `if` chain at all.
```

File: rules/S3923/scala/rule.adoc [hunk 5]
```diff
@@ -1,8 +1,6 @@
 == Why is this an issue?
 
-Having all branches in a ``++match++`` or ``++if++`` chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be a ``++match++``/``++if++`` chain at all.
-
-=== Noncompliant code example
+Having all branches in a `match` or `if` chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be a `match`/`if` chain at all.
```
Comment (amelie-renard-sonarsource) [inline rules/S3923/scala/rule.adoc:3]:
```suggestion
Having all branches in a `match` or `if` chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be a `match` or `if` chain at all.
```

File: rules/S3923/tsql/rule.adoc [hunk 6]
```diff
@@ -1,8 +1,6 @@
 == Why is this an issue?
 
-Having all branches in a ``++CASE++``, ``++IF++`` or ``++IIF++`` chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be a ``++CASE++``/``++IF++``/``++IIF++`` chain at all.
-
-=== Noncompliant code example
+Having all branches in a `CASE`, `IF` or `IIF` chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be a `CASE`/`IF`/`IIF` chain at all.
```
Comment (amelie-renard-sonarsource) [inline rules/S3923/tsql/rule.adoc:3]:
```suggestion
Having all branches in a `CASE`, `IF` or `IIF` chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be a `CASE`, `IF`, or `IIF` chain at all.
```

File: rules/S3923/tsql/rule.adoc [hunk 7]
```diff
@@ -25,7 +23,7 @@ SELECT
 
 === Exceptions
 
-This rule does not apply to ``++IF++``/``++CASE++`` chains without ``++ELSE++`` clauses.
+This rule does not apply to `IF`/`CASE` chains without `ELSE` clauses.
```
Comment (amelie-renard-sonarsource) [inline rules/S3923/tsql/rule.adoc:26]:
```suggestion
This rule does not apply to `IF` or `CASE` chains without `ELSE` clauses.
```

File: rules/S3923/vbnet/rule.adoc [hunk 8]
```diff
@@ -1,20 +1,18 @@
 == Why is this an issue?
 
-Having all branches in a ``++Select Case++`` or ``++If++`` chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be a ``++Select Case++`` / ``++If++`` chain at all.
-
-=== Noncompliant code example
+Having all branches in a `Select Case` or `If` chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be a `Select Case` / `If` chain at all.
```
Comment (amelie-renard-sonarsource) [inline rules/S3923/vbnet/rule.adoc:3]:
```suggestion
Having all branches in a `Select Case` or `If` chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be a `Select Case` or `If` chain at all.
```

**Review by amelie-renard-sonarsource — CHANGES_REQUESTED** [review]

File: rules/S3923/abap/rule.adoc (hunk 2 above)
Comment (frederic-tingaud-sonarsource) [reply rules/S3923/abap/rule.adoc:7]: done

File: rules/S3923/kotlin/rule.adoc [hunk 9]
```diff
@@ -1,8 +1,6 @@
 == Why is this an issue?
 
-Having all branches in a ``++when++`` or ``++if++`` chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be a ``++when++``/``++if++`` chain at all.
-
-=== Noncompliant code example
+Having all branches in a `when` or `if` chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be a `when`/`if` chain at all.
```
Comment (amelie-renard-sonarsource) [inline rules/S3923/kotlin/rule.adoc:3]: This needs also to be updated

File: rules/S3923/plsql/rule.adoc [hunk 10]
```diff
@@ -1,8 +1,6 @@
 == Why is this an issue?
 
-Having all branches in a ``++CASE++`` or ``++IF++``/``++ELSIF++`` chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be a ``++CASE++``/``++IF++``/``++ELSIF++`` chain at all.
-
-=== Noncompliant code example
+Having all branches in a `CASE` or `IF`/`ELSIF` chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be a `CASE` or `IF`/`ELSIF` chain at all.
```
Comment (amelie-renard-sonarsource) [inline rules/S3923/plsql/rule.adoc:3]: This needs also to be updated

File: rules/S3923/python/rule.adoc [hunk 11]
```diff
@@ -1,8 +1,6 @@
 == Why is this an issue?
 
-Having the same implementation in every branch of an ``++if++`` chain is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be an ``++if++`` chain at all.
-
-=== Noncompliant code example
+Having the same implementation in every branch of an `if` chain is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be an `if` chain at all.
```
Comment (amelie-renard-sonarsource) [inline rules/S3923/python/rule.adoc:3]: This needs also to be updated

File: rules/S3923/ruby/rule.adoc [hunk 12]
```diff
@@ -1,8 +1,6 @@
 == Why is this an issue?
 
-Having all branches in a ``++case++`` or ``++if++`` chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be a ``++case++``/``++if++`` chain at all.
-
-=== Noncompliant code example
+Having all branches in a `case` or `if` chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be a `case` or `if` chain at all.
```
Comment (amelie-renard-sonarsource) [inline rules/S3923/ruby/rule.adoc:3]: This needs also to be updated

File: rules/S3923/scala/rule.adoc [hunk 13]
```diff
@@ -1,8 +1,6 @@
 == Why is this an issue?
 
-Having all branches in a ``++match++`` or ``++if++`` chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be a ``++match++``/``++if++`` chain at all.
-
-=== Noncompliant code example
+Having all branches in a `match` or `if` chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be a `match` or `if` chain at all.
```
Comment (amelie-renard-sonarsource) [inline rules/S3923/scala/rule.adoc:3]: This needs also to be updated

File: rules/S3923/tsql/rule.adoc [hunk 14]
```diff
@@ -1,8 +1,6 @@
 == Why is this an issue?
 
-Having all branches in a ``++CASE++``, ``++IF++`` or ``++IIF++`` chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be a ``++CASE++``/``++IF++``/``++IIF++`` chain at all.
-
-=== Noncompliant code example
+Having all branches in a `CASE`, `IF` or `IIF` chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be a `CASE`, `IF`, or `IIF` chain at all.
```
Comment (amelie-renard-sonarsource) [inline rules/S3923/tsql/rule.adoc:3]: This needs also to be updated

**Review by amelie-renard-sonarsource — CHANGES_REQUESTED** [review]

File: rules/S3923/abap/rule.adoc (hunk 1 above)
Comment (frederic-tingaud-sonarsource) [reply rules/S3923/abap/rule.adoc]: done

File: rules/S3923/kotlin/rule.adoc (hunk 9 above)
Comment (frederic-tingaud-sonarsource) [reply rules/S3923/kotlin/rule.adoc]: Fixed for all, sorry for the noise

**Review by amelie-renard-sonarsource — APPROVED** [review]

**Pull request merged**